69.md

```diff
@@ -0,0 +1,293 @@
+NIP-69
+======
+
+Nostr Offer STRings
+-------------------
+
+`rfc` `optional`
+
+This NIP proposes a format for static payment codes in Nostr as a successor to LNURL-Pay, enabling users to initiate Lightning Network payments by scanning or clicking a string.
+
+## Motivation
+
+Reliance on LNURL has led to centralization via custodial solutions due to the legacy baggage of IP4-NAT/Domain/SSL requirements. Nostr's use-cases are already centered around Lightning, whether as a wallet connector, zap receipts, or its overlapping network effects, rendering Nostr as a de facto "3rd layer" for Lightning and a natural successor to LNURL.
+
+When layered over kind 21000, NWC, or other potential Lightning RPCs, this specification enables a seamless user experience similar to legacy LNURL-Pay, without the domain and SSL requirements. Additionally, the signed nature of Nostr communications minimizes the trust requirements inherent in LNURL when a receiving node must outsource the serving of web requests.
+
+## Specification
+
+### Static Payment Code Format
+
+The static payment code is a bech32 (per [NIP-19](19.md)) encoded string prefixed with `noffer`. The encoded string will include the following TLV (Type-Length-Value) items:
+
+- `0`: The 32 bytes of the receiving service's public key, encoded in hex.
+- `1`: The relay URL where the receiving service subscribes to payment requests.
+- `2`: The offer identifying string.
+- `3`: A flag indicating the pricing type:
+  - `0`: Fixed price (Price stated in 4 will not change)
+  - `1`: Variable price (Price must be calculated and will be reflected in the invoice, useful for products priced in fiat)
+  - `2`: Spontaneous payment (Payer specifies the amount in request payload)
+- `4`: The price in sats (optional for display purposes).
```

Comment:
Afaict, this means that one can only specify invoices in sats. It would be nice if you could consider to use msats here. Explanation:

[LUD-06](https://github.com/lnurl/luds/blob/luds/06.md), [NIP-47](https://github.com/nostr-protocol/nips/blob/master/47.md) and [NIP-57](https://github.com/nostr-protocol/nips/blob/master/57.md) also specify the amount in msats. It would also be good for us at [Stacker News](https://stacker.news) if this spec supports msats out of the box because we forward payments between users and take 10% as a fee for sybil resistance and rewards. This means that if someone zaps 25 sats, 22.5 sats will be received which requires being able to specify invoices in msats.

Ideally would be amount + unit like [BOLT11](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md) though but msats is probably good enough and would be consistent with other specs as mentioned.

I have to mention thought that most LNURL providers seem to simply round to nearest sat so don't really support msats; contrary to the spec.

Reply:
> LNURL providers seem to simply round to nearest sat 

Yea I considered this but steered away from assuming any baggage from LNURL... the issue with millisats is that they can't actually be settled in payments (technically they are shitcoins and thus rounding happens regardless of whether we use them or not). The scope operates on the assumption that offers will be used in sales reports and things of that nature which do not go to decimal places of a penny etc and so I didn't re-introduce that here.